olp-cpp-sdk-core/src/network2/ios/OLPNetworkIOS.h

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <mutex>
+
+#include "olp/core/network2/Network.h"
+
+#ifdef __OBJC__
```

Comment:
I removed the `HttpClientPtr` there, cause it is not needed. Please, check again.

We don't need to have a `std::unique_ptr`, because `HttpClient`is already the obj-c object, and its lifetime is managed by the ARC.

Reply:
Ok, `HttpClientPtr` also saves us from future changes in choosing pointer for that data member. Maybe we indeed will switch to smart pointers.